From d5c78b3abbfbccb9919be8600741409f16d0846e Mon Sep 17 00:00:00 2001
From: Alexander Grund <Flamefire@users.noreply.github.com>
Date: Wed, 22 Oct 2025 09:03:16 +0200
Subject: [PATCH] Check all environment variables for Slurm environment

In some environments only the SLURM_JOB_ID might be set, e.g. when using hooks for SSH to a node with an existing allocation

This causes a false positive in the detection and later `KeyError` on e.g. `SLURM_LOCALID`
---
 jax/_src/clusters/slurm_cluster.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/jax/_src/clusters/slurm_cluster.py b/jax/_src/clusters/slurm_cluster.py
index 8cec07601094..a8bb1b8a287f 100644
--- a/jax/_src/clusters/slurm_cluster.py
+++ b/jax/_src/clusters/slurm_cluster.py
@@ -30,7 +30,8 @@ class SlurmCluster(clusters.ClusterEnv):
 
   @classmethod
   def is_env_present(cls) -> bool:
-    return _JOBID_PARAM in os.environ
+    return all(var in os.environ for var in
+               (_JOBID_PARAM, _NODE_LIST, _PROCESS_COUNT, _PROCESS_ID, _LOCAL_PROCESS_ID))
 
   @classmethod
   def get_coordinator_address(cls, timeout_secs: int | None) -> str: